sonic-thermalctld/scripts/thermalctld

```diff
@@ -315,16 +293,18 @@ class FanUpdater(logger.Logger):
         fan_status = self.fan_status_dict[fan_name]
 
         speed = NOT_AVAILABLE
-        speed_tolerance = NOT_AVAILABLE
         speed_target = NOT_AVAILABLE
+        is_under_speed = NOT_AVAILABLE
+        is_over_speed = NOT_AVAILABLE
         fan_fault_status = NOT_AVAILABLE
         fan_direction = NOT_AVAILABLE
         is_replaceable = try_get(fan.is_replaceable, False)
         presence = try_get(fan.get_presence, False)
         if presence:
             speed = try_get(fan.get_speed)
-            speed_tolerance = try_get(fan.get_speed_tolerance)
             speed_target = try_get(fan.get_target_speed)
+            is_under_speed = try_get(fan.is_under_speed)
```

Comment:
Is it better to check if this function is not implemented and fall back to old code than putting an implementation in the base class (which one would not understand unless he knew about this change) esp because the changes span across repos.

Reply:
Having a default implementation in the base class allows for the same calculation to be used across other code, such as `sonic-mgmt` tests. If we fall back to a hard-coded calculation inside thermalctld, then the calculation must also be hard-coded anywhere else fan under/over speed conditions should be checked. Existing `sonic-mgmt` speed tolerance checks use different (and incorrect) calculation logic, so having a default API implementation solves this issue.